PySide2 compatibility

MolotovCherry opened the pull request:
Only minor changes needed to fix errors with PySide2

Mostly the parameters to certain functions changed and need to be handled differently between API's

- Not 100% sure of the impact, but it works 100% fine for me, never noticed anything different. Please review it though

MolotovCherry committed:
PySide2 compatibility

klauer commented on qtpydocking/dock_container_widget.py:
```diff
@@ -227,8 +228,14 @@ def drop_into_container(self,
 
-        new_dock_areas = floating_dock_container.findChildren(
-            DockAreaWidget, '', Qt.FindChildrenRecursively)
+        api = os.environ['QT_API']
```
I think the preferred check here may be to use `qtpy.API`, which tells you exactly which one qtpy chose after its logic (see [here](https://github.com/spyder-ide/qtpy/blob/275fe456dc7095162dbd2054a1d866bc28eefacc/qtpy/__init__.py#L216)).

MolotovCherry replied:
Fixed. Also, rather than using `qtpy.API`, I used the specific `boolean` variables they provided (much more specific)

klauer commented on qtpydocking/dock_container_widget.py:
```diff
@@ -227,8 +228,14 @@ def drop_into_container(self,
+            new_dock_areas = floating_dock_container.findChildren(
+                DockAreaWidget, '')
+        else:
+            new_dock_areas = floating_dock_container.findChildren(
```
Since we need to use the API check to get `findChildren` to work multiple times, would you mind making a new method in `qtpydocking.util` that does this check only once? e.g., `util.find_children(parent, cls, name='', options=Qt.FindChildrenRecursively)`

MolotovCherry replied:
Added. Much more cohesive now. :)

klauer reviewed (commented):
Thanks so much for testing this with PySide2! I need to get around to having Travis CI working with both PySide2/PyQt5 environments at some point.

If you don't have time to make the requested changes, I'm OK with merging for now - just let me know.

MolotovCherry committed:
Add API functions to util

MolotovCherry commented:
Fixed. I used the specific API boolean variables qtpy provided for more specificity. I also fixed some silent bugs. Pyside and PyQt4, in addition to PySide2, all do not support `Qt.FindChildOptions`. Only PyQt5 supports `Qt.FindChildOptions`. The program (probably) would've crashed when it ran into any of these in those API's.

klauer reviewed (approved):
Looks great! Thanks again. :+1:

Ran the tests locally and verified things are still working.
